bloom/commands/git/import_upstream.py

```diff
@@ -38,6 +38,7 @@
 import tarfile
 import tempfile
 
+from io import open
```

Comment:
> is this import necessary if we don't need to pass an ecoding to open?

I know that I only worked on this a couple hours ago but I've already forgotten. If it is needed it will be because of the different types expected by the `open` builtin versus `io.open`. I'll revert this change and retest to be sure.



> I guess in that case we can do the same here so that "open" always refers to the builtin open and io.open to the fancier version.

This is a trade-off between having one consistent API for `open` and only using `io.open` when the python2 builtin doesn't work. `io.open` has the same API in Python 2 and Python 3 and is encoding aware.

Using it everywhere would cut down on some mental overhead and would future-proof code against utf-8 characters at the cost of requiring utf-8 handling everywhere. If we only use `io.open` where we know that we need it we avoid that overhead but risk future errors when utf-8 pops up where there was only ascii previously.

That being said, future errors are speculative and mental overhead is a surety so it's really a question of which is more work for the programmer: Sometimes using `io.open` or always using `io.open`?